src/controllers/admin/mentor.controller.ts

```diff
@@ -0,0 +1,37 @@
+import type { Request, Response } from 'express'
+import { updateMentorStatus } from '../../services/admin/mentor.service'
+import { ApplicationStatus, ProfileTypes } from '../../enums'
+import type Profile from '../../entities/profile.entity'
+
+export const mentorStatusHandler = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    const user = req.user as Profile
+    const { status } = req.body
+    const { mentorId } = req.params
+
+    if (!(status.toUpperCase() in ApplicationStatus)) {
+      res.status(400).json({ message: 'Please provide a valid status' })
+      return
+    }
+
+    if (user.type !== ProfileTypes.ADMIN) {
```

Comment:
I think it's better to do this validation before the "valid status" validation

Reply:
oh yeah